test/parallel/test-fs-rmdir-recursive.js

```diff
@@ -105,6 +100,25 @@ function removeAsync(dir) {
   dir = nextDirPath();
   makeNonEmptyDirectory(1, 10, 2, dir, true);
   removeAsync(dir);
+
+  // Should fail if target does not exist
+  fs.rmdir(
+    path.join(tmpdir.path, 'noexist.txt'),
+    { recursive: true },
+    common.mustCall((err) => {
+      assert.strictEqual(err.code, 'ENOENT');
+    })
+  );
+
+  // Should fail if target is a file
+  const filePath = path.join(tmpdir.path, 'rmdir-async-file.txt');
+  fs.writeFileSync(filePath, '');
+  fs.rmdir(filePath, { recursive: true }, common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_INVALID_ARG_VALUE');
+    assert.strictEqual(err.name, 'TypeError');
+    assert.match(err.message, /^The argument 'path' is not a directory\./);
+  }));
+  fs.unlinkSync(filePath);
```

Comment:
imo this should be guaranteed to be called--not sure if it is. also: don't know offhand if other tests try to guarantee this or just leave crap all over the fs

Reply:
I can use a try/finally to clean this up.